Chirich/Lab1/Program.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Crc32;
+
+namespace Lab1
+{
+    internal class Program
+    {
+        private static readonly Dictionary<string, HashAlgorithm> Algorithms = new Dictionary<string, HashAlgorithm>
+        {
+            {"sha1", new SHA1Managed()},
+            {"sha256", new SHA256Managed()},
+            {"md5", new SHA256Managed()},
+            {"crc32", new Crc32Algorithm()}
+        };
+
+        private static void Main(string[] args)
+        {
+            if (args.Length == 0 || args.Length == 1 && args[0] == "help")
+            {
+                Help();
+                return;
+            }
+
+            var filePath = args.Length >= 1 ? args[0] : string.Empty;
+            var blockSize = 1024;
+            if (args.Length >= 2) int.TryParse(args[1], out blockSize);
+            var algorithType = args.Length >= 3 ? args[2].ToLower() : "sha1";
+            var mode = args.Length >= 4 ? args[3].ToLower() : "sync";
+
+            Console.WriteLine($"Params: File:{filePath} Block:{blockSize} Algorithm: {algorithType} Mode: {mode}");
+
+
+
+            if (!Algorithms.ContainsKey(algorithType))
+            {
+                Console.WriteLine($"Algorithm {algorithType} not found.");
+                return;
+            }
+
+            if (mode != "sync" && mode != "async")
+            {
+                Console.WriteLine($"Mode {mode} not found. Mode should be sync or async");
+                return;
+            }
+
+
+            Console.Write(" Starting....");
+            Console.WriteLine($"{filePath} {blockSize} {algorithType} {mode}");
+
+            HashFile(filePath, algorithType, blockSize, mode == "sync");
+        }
+
+        private static void Help()
+        {
+            Console.WriteLine("Lab for hashing files. \n");
+            Console.WriteLine("Parms: ");
+            Console.WriteLine(" 1) File path.");
+            Console.WriteLine(" 2) Block size - integer form 1024 to 1048576 as 2^n");
+            Console.Write(" 3) Algorithms. one of ");
+            Algorithms.Keys.ToList().ForEach(item => Console.Write($"{item} "));
+            Console.WriteLine();
+            Console.WriteLine(" 4) Mode - sync or async");
+        }
+
+        private static void HashFile(string inputPath, string algorithType, int blockSize, bool isSync)
+        {
+            var readKey = new object();
+            var writeKey = new object();
+
+            var outputPath = $"{inputPath}.hashes_{algorithType}";
+            var algorithm = Algorithms[algorithType];
+            var threadsAmount = isSync ? 1 : Environment.ProcessorCount;
+
+            var wrWatch = new Stopwatch();
```

Comment:
Well... Well... Well...
Looks like a complete copy of Vlad's task... 
https://github.com/PabloDeCortes/hashingtask/blob/vladislav_karpovich/Karpovich/HashingFileLab/HashingFileLab/HashService.cs#L35

Not good.

Reply:
Also, you can check what was said about write speed in Vlad's Pull Request - it was not needed to calculate write speed, it was needed to calculate hash speed, the measurement should be in another place.